homework/java/shelest_vladimir/login_server/src/test/java/by/jrr/user/UserTest.java

```diff
@@ -0,0 +1,89 @@
+package by.jrr.user;
+
+import by.jrr.bean.User;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Objects;
+
+public class UserTest {
+    @Test
+    public void toString_Test() {
+        User user = new User("Login", "Password");
+        String expected = "User{" +
+                "login = '" + "Login" + '\'' +
+                ", password = '" + "Password" + '\'' +
+                ", attempt = " + "3" +
+                ", is blocked = " + "false" +
+                '}';
+        String actual = user.toString();
+
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void equals_Test() {
+        User user = new User("Login", "Password");
+        User anotherUser = new User("Login", "AnotherPassword");
+
+        boolean expected = true;
+        boolean actual = user.equals(anotherUser);
+
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void equals_AnotherUser_Test() {
+        User user = new User("Login", "Password");
+        User anotherUser = new User("AnotherLogin", "Password");
+
+        boolean expected = false;
+        boolean actual = user.equals(anotherUser);
+
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void equalsWithSameUser_Test() {
+        User user = new User("Login", "Password");
+
+        boolean expected = true;
+        boolean actual = user.equals(user);
+
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void failedEquales_NullOrOtherType_Test() {
+        User user = new User("Login", "Password");
+        boolean expected = false;
+
+        boolean actual = user.equals(null);
```

Comment:
Our static analysis engine detects that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [EC: Call to equals(null) (EC_NULL_ARG)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#ec-call-to-equals-null-ec-null-arg).
This method calls equals(Object), passing a null value as the argument. According to the contract of the equals() method, this call should always return false.